cmd/kubernetes-admission-controller/anchore/anchore_api_image_backend.go

```diff
@@ -103,53 +108,46 @@ func (p APIImageBackend) Analyze(asUser Credential, imageReference string) error
 func (p APIImageBackend) DoesPolicyCheckPass(asUser Credential, imageDigest, imageTag, policyBundleID string) (bool, error) {
 	ctx := authContextFromCredential(asUser)
 	var err error
-	var evaluations []interface{}
+	var evaluation interface{}
 	if policyBundleID != "" {
-		evaluations, _, err = p.client.GetImagePolicyCheck(ctx, imageDigest).
+		evaluation, _, err = p.client.GetImagePolicyCheckByDigest(ctx, imageDigest).
 			Tag(imageTag).
 			Interactive(true).
 			PolicyId(policyBundleID).
 			Execute()
 	} else {
-		evaluations, _, err = p.client.GetImagePolicyCheck(ctx, imageDigest).Tag(imageTag).Interactive(true).Execute()
+		evaluation, _, err = p.client.GetImagePolicyCheckByDigest(ctx, imageDigest).Tag(imageTag).Interactive(true).Execute()
 	}
 	if err != nil {
 		return false, err
 	}
 
-	if len(evaluations) > 0 {
-		var resultStatus string
-		if resultMap, ok := evaluations[0].(map[string]interface{}); ok {
-			resultStatus = getPolicyEvaluationStatus(resultMap)
-		}
-		return strings.ToLower(resultStatus) == "pass", nil
+	var resultStatus string
+	resultMap, ok := evaluation.(anchoreEnterprise.PolicyEvaluation)
+	if ok {
+		resultStatus = getPolicyEvaluationStatus(resultMap.Evaluations)
 	}
-
-	return false, nil
+	return strings.ToLower(resultStatus) == "pass", nil
```

Comment:
is this ok if resultStatus has not been initialized?

Reply:
Yup strings are initialized on declaration as empty string "" in go